### PR TITLE
feat: add unknown severity handling across analysis paths

### PR DESCRIPTION
**Pull request by a-oren**
## Summary

- Add `unknown` severity field to `ResponseMetrics` interface and map it in both component analysis (`analysis.ts`) and stack analysis (`rhda.ts`) code paths
- Update image analysis `getHighestSeverity()` cascade to handle `unknown`, ranked below LOW (CRITICAL > HIGH > MEDIUM > LOW > UNKNOWN)
- Fix pre-existing bug: `medium` severity was incorrectly mapped from `sourceData.summary.dependencies` instead of `sourceData.summary.medium`
- Bump `@trustify-da/trustify-da-api-model` to 2.0.9 and `@trustify-da/trustify-da-javascript-client` to 0.3.0-ea.4227d87 for the updated `SourceSummary` model

Implements [TC-4727](https://redhat.atlassian.net/browse/TC-4727)

## Test plan

- [x] Existing tests pass (288 passing, 3 pre-existing failures unrelated to this change)
- [x] New test: `getHighestSeverity` returns `UNKNOWN` when only unknown > 0
- [x] New test: `getHighestSeverity` returns `LOW` when both low and unknown > 0 (LOW outranks UNKNOWN)
- [x] New test: `getHighestSeverity` returns `NONE` when no severities present
- [x] Lint passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-4727]: https://redhat.atlassian.net/browse/TC-4727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Handle unknown vulnerability severity across dependency and image analysis paths and update supporting models and tests.

New Features:
- Add support for an unknown severity field in analysis response metrics and image analysis highest-severity calculation.

Bug Fixes:
- Correct the mapping of medium severity counts from the source summary in dependency analysis.

Enhancements:
- Update metrics extraction for dependency and stack analysis to include unknown severity counts.
- Upgrade Trustify API model and JavaScript client dependencies to consume the updated SourceSummary model.

Tests:
- Add image analysis tests covering UNKNOWN, LOW vs UNKNOWN precedence, and NONE highest-severity outcomes.

### CODE REVIEW

File: test/imageAnalysis/analysis.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import { type IOptions } from '../../src/imageAnalysis';
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+/**
+ * Creates a mock IExhortAnalysisReport with a single provider/source
+ * using the given SourceSummary fields.
+ */
+function createMockReport(imageRef: string, summary: Record<string, number>) {
```
[the remark this answers is not in the document]
Comment (a-oren) [reply test/imageAnalysis/analysis.test.ts]: [sdlc-workflow/verify-pr] Classified as **suggestion** — no sub-task created.

The suggestion to use `Partial<SourceSummary>` instead of `Record<string, number>` was evaluated for convention upgrade but no matching convention was found in CONVENTIONS.md or codebase precedent. It remains an optional suggestion.

**Review by a-oren — COMMENTED** [review]
## Verification Report — TC-4727

| Check | Verdict | Details |
|---|---|---|
| Scope Containment | ✅ PASS | All 3 task files modified; out-of-scope files justified (package deps, test file) |
| Diff Size | ✅ PASS | 132 insertions, 8 deletions across 6 files |
| Commit Traceability | ✅ PASS | Commit 76a1feb references "Implements TC-4727" |
| Sensitive Patterns | ✅ PASS | No secrets or credentials detected |
| CI Status | ✅ PASS | All CI checks pass (build-and-test, Sourcery, main) |
| Acceptance Criteria | ✅ PASS | All 5 acceptance criteria verified in code |
| Verification Commands | ✅ PASS | Lint passes; 3 test failures are pre-existing |
| Test Quality | ⚠️ WARN | Tests could be parameterized; `///` used instead of `/** */` for doc comments |
| Test Change Classification | ℹ️ ADDITIVE | New test file added |
| Review Feedback | ✅ PASS | 1 suggestion (no code change requests) |
| Root-Cause Investigation | N/A | No sub-tasks created |

### Overall: ✅ PASS

**Test Quality details:**
- *Repetitive Test Detection:* 3 tests share identical setup/execution pattern — could use parameterized approach
- *Test Documentation:* Tests use `///` triple-slash comments instead of JSDoc `/** */` convention
- *Eval Quality:* N/A (no eval infrastructure in this repository)

These are informational warnings and do not block merge.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.9.2.*

**Review by Strum355 — APPROVED** [review]
